tests/upgrades/test_containers_removal.py

```diff
@@ -0,0 +1,209 @@
+            tag=u'docker'
+        )
+        docker_host.create()
+        try:
+            docker_host.install_katello_ca()
+
+            compute_resource = entities.DockerComputeResource(
+                name=compute_resource_name,
+                organization=[org],
+                url='http://{0}:2375'.format(docker_host.ip_addr),
+            ).create()
+
+            # Only one registry with given URL can exist on Satellite,
+            # so search for it first and create it only if necessary
+            try:
+                registry = entities.Registry().search(filters={'url': registry_url})[0]
+            except IndexError:
+                registry = entities.Registry(
+                    url=registry_url,
+                    organization=[org],
+                ).create()
+
+            # container from dockerhub
+            dockerhub_container = entities.DockerHubContainer(
+                command='sleep inf',
+                compute_resource=compute_resource,
+                organization=[org],
+            ).create()
+            self.assertEqual(
+                dockerhub_container.compute_resource.id, compute_resource.id)
+
+            # container from external registry
+            external_container = entities.DockerRegistryContainer(
+                command='sleep inf',
+                compute_resource=compute_resource,
+                organization=[org],
+                registry=registry,
+                repository_name=repo_name,
+            ).create()
+            self.assertEqual(
+                external_container.compute_resource.id, compute_resource.id)
+            self.assertEqual(external_container.registry.id, registry.id)
+            self.assertEqual(external_container.repository_name, repo_name)
+
+            running_containers = docker_host.run('docker ps')
+            self.assertEqual(running_containers.return_code, 0)
+
+            self.assertTrue(any(dockerhub_container.name in line
+                                for line in running_containers.stdout))
+            self.assertTrue(any(external_container.name in line
+                                for line in running_containers.stdout))
+
+            scenario_dict = {self.__class__.__name__: {
+                'docker_host': docker_host.hostname,
+                'dockerhub_container': dockerhub_container.name,
+                'external_container': external_container.name,
+            }}
+            create_dict(scenario_dict)
+        except Exception as exp:
+            self._vm_cleanup(hostname=docker_host.hostname)
+            raise Exception(exp)
+
+    @post_upgrade(depend_on=test_pre_scenario_containers_support_removal)
+    def test_post_scenario_containers_support_removal(self):
+        """Post-upgrade scenario test to verify containers created and run
+        before upgrade are still running after upgrade.
+
+        :id: preupgrade-f6de07ae-14c7-4452-9cb1-cafe2aa648ae
```

Comment:
should be post-upgrade

Reply:
Good catch! Changed in b30046f687c0a0de2e701ec29210706339f411e5